symla/kronecker.py

```diff
@@ -46,8 +43,17 @@ def name(self):
         return self._name
 
     @property
-    def shape(self):
-        return self._shape
+    def dimension(self):
+        return self._dimension
+
+    @property
+    def field(self):
+        return self._field
+    
+    @property
+    def dtype(self):
```

Comment:
For code generation, the data type can be infered from the field of the vector space if this information is not provided